test/read.jl

```diff
@@ -0,0 +1,245 @@
+mktempdir() do dir
+
+tasks = []
+
+# Create test file...
+filename = joinpath(dir, "file.txt")
+text = "C1,C2\n1,2\na,b\n"
+
+# List of IO producers...
+l = Vector{Tuple{AbstractString,Function}}()
+
+
+# File
+io = (text) -> begin
+    open(io-> write(io, text), filename, "w")
+    Base.Filesystem.open(filename, Base.Filesystem.JL_O_RDONLY)
+end
+s = io(text)
+@test isa(s, IO)
+@test isa(s, Base.Filesystem.File)
+close(s)
+push!(l, ("File", io))
+
+
+# IOStream
+io = (text) -> begin
+    open(io-> write(io, text), filename, "w")
+    open(filename)
+end
+s = io(text)
+@test isa(s, IO)
+@test isa(s, IOStream)
+close(s)
+push!(l, ("IOStream", io))
+
+
+# IOBuffer
+io = (text)->IOBuffer(text)
+s = io(text)
+@test isa(s, IO)
+@test isa(s, IOBuffer)
+close(s)
+push!(l, ("IOBuffer", io))
+
+
+function run_test_server(srv, text)
+    push!(tasks, @async begin
+        try
+            sock = accept(srv)
+            try
+                write(sock,text)
+            catch e
+                if typeof(e) != Base.UVError
+                    rethrow(e)
+                end
+            finally
+                close(sock)
+            end
+        finally
+            close(srv)
+        end
+    end)
+end
+
+if !haskey(ENV, "TRAVIS")
```

Comment:
I hope that [this `yield()`](https://github.com/samoconnor/julia/commit/490cee37d8962d01dfabbcc719ff4062be3bb1cc#diff-435745fdfa02a0dc4f3b747ba977af11R63) will fix the deadlock. Waiting for Travis to start up to find out...